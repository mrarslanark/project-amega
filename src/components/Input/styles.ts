import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  wrapper: {
    rowGap: 4,
  },
  input: {
    backgroundColor: 'rgba(0, 0, 0, 0.05)',
    padding: 16,
    borderRadius: 12,
    color: 'black',
  },
  errorWrapper: {
    alignItems: 'flex-end',
  },
});
