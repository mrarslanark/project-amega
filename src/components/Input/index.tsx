import React from 'react';
import {TextInput, TextInputProps, View} from 'react-native';
import {styles} from './styles';
import Text from '../Text';

interface InputProps extends TextInputProps {
  error: string | null;
}

const Input: React.FC<InputProps> = ({error, ...props}): React.JSX.Element => {
  return (
    <View style={styles.wrapper}>
      <TextInput
        {...props}
        style={styles.input}
        placeholderTextColor={'gray'}
      />
      {error && error.length > 0 && (
        <View style={styles.errorWrapper}>
          <Text variant="error">{error}</Text>
        </View>
      )}
    </View>
  );
};

export default Input;
