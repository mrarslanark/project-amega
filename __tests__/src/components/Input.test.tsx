import React from 'react';
import {render, screen} from '@testing-library/react-native';
import Input from '../../../src/components/Input';

describe('Input', () => {
  it('should render successfully', () => {
    render(<Input error={null} />);
    expect(screen).toMatchSnapshot();
  });

  it('should render successfully with an error message', () => {
    const errorText = 'Error occured';
    render(<Input error={errorText} />);

    const errorTextView = screen.getByText(errorText);
    expect(errorTextView).toHaveTextContent(errorText);

    expect(screen).toMatchSnapshot();
  });
});
