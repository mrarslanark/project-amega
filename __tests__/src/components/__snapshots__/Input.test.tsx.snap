// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render successfully 1`] = `
<View
  style={
    {
      "rowGap": 4,
    }
  }
>
  <TextInput
    placeholderTextColor="gray"
    style={
      {
        "backgroundColor": "rgba(0, 0, 0, 0.05)",
        "borderRadius": 12,
        "color": "black",
        "padding": 16,
      }
    }
  />
</View>
`;

exports[`Input should render successfully with an error message 1`] = `
<View
  style={
    {
      "rowGap": 4,
    }
  }
>
  <TextInput
    placeholderTextColor="gray"
    style={
      {
        "backgroundColor": "rgba(0, 0, 0, 0.05)",
        "borderRadius": 12,
        "color": "black",
        "padding": 16,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "flex-end",
      }
    }
  >
    <Text
      style={
        {
          "color": "#CE4760",
          "fontSize": 14,
          "fontWeight": "500",
        }
      }
    >
      Error occured
    </Text>
  </View>
</View>
`;
